crates/api_common/src/utils.rs

```diff
@@ -592,3 +594,29 @@ pub async fn delete_user_account(person_id: PersonId, pool: &DbPool) -> Result<(
 
   Ok(())
 }
+
+pub async fn listing_type_with_site_default(
+  listing_type: Option<ListingType>,
+  pool: &DbPool,
+) -> Result<ListingType, LemmyError> {
+  Ok(match listing_type {
+    Some(l) => l,
+    None => {
+      let site = blocking(pool, Site::read_local_site).await??;
+      ListingType::from_str(&site.default_post_listing_type)?
+    }
+  })
```

Comment:
Should be `unwrap_or()` (but maybe that doesnt work because of async).

Reply:
What should be unwrap_or() here? The site default_post_listing_type should already have a good default in SQL.